cql3/selection/selection.cc

```diff
@@ -386,17 +388,61 @@ std::unique_ptr<result_set> result_set_builder::build() {
     return std::move(_result_set);
 }
 
-result_set_builder::restrictions_filter::restrictions_filter(::shared_ptr<restrictions::statement_restrictions> restrictions,
+namespace {
+
+using namespace expr;
+
+/// A visitor that copies an expression, but replaces token operator and unselected-column operators with the
+/// TRUE literal.
+struct drop_token_and_unselected_columns {
+    const selection& sel;
+
+    expression operator()(const conjunction& c) const {
+        auto r = c.children | std::views::transform([this] (const expression& e) { return std::visit(*this, e); });
+        return conjunction{std::vector<expression>(r.begin(), r.end())};
```

Comment:
I am filled with indignation that a vector(const auto std::range) consructor doesn't exist. We could then write 

```c++
    return conjunction{c.children | std::views::transform([this] (const expression& e) { return std::visit(*this, e); })};
```

Reply:
We could even have written

```c++
    return conjunction{c.children | std::views::transform(std::bind_front(std::visit, std::ref(*this)))};
```

Except that std::visit is a template and so a second-class object. We'd be getting close to 5X bloat from the equivalent Haskell or lisp.

Reply:
We wouldn't even get to code generation, because bind_front can't pick the right overload: https://godbolt.org/z/ozPvad

Reply:
Yes, because it's a template, as I mentioned.